File: solutions/clojure/acronym/1/src/acronym.clj
(ns acronym
   (:require [clojure.string :as str]))

(defn only-letters [s]
 (apply str (filter #(Character/isLetter %) s))
  )

(defn uppercase-first [s]
  (apply str (concat (str/upper-case (first s)) (rest s))))

(defn process [s]
  (if (= s (str/upper-case s)) (first s) 
    (apply str (filter #(Character/isUpperCase %) (seq (uppercase-first s))))))

(defn acronym  [phrase]
  (if (empty? phrase) "" 
 (str/join (map process (map only-letters (str/split (str/replace phrase "-" " " )  #"\s+")))))
  )

File: solutions/julia/acronym/1/acronym.jl
function acronym(phrase)
    out = ""
    for i in 1:length(phrase)        
        if 1==i || phrase[i-1] in [' ', '-', '_']
            c = phrase[i]
            if isletter(c) 
                out *= uppercase(c)
            end
        end
    end
    out
end
